saleor/checkout/complete_checkout.py

```diff
@@ -180,15 +180,23 @@ def _process_shipping_data_for_order(
     delivery_method_info = checkout_info.delivery_method_info
     shipping_address = delivery_method_info.shipping_address
 
-    if (
-        delivery_method_info.store_as_customer_address
-        and checkout_info.user
-        and shipping_address
+    # allow saving shipping address in case the delivery method is not provided,
+    # but the shipping address is set
+    if not delivery_method_info.delivery_method:
+        shipping_address = checkout_info.shipping_address
+
+    if _should_store_shipping_address_in_user_addresses(
```

Comment:
Not sure if this should be a part of this feature or separate issue, but shouldn't we move this part to on_commit action of checkout-complete? 
Right now, for plugins flow, the address can be created even if we raise an exception (plugin flow is split into two transaction blocks. This happens in first transaction block. So in case of having an exception in the second block, we will make a store-address action, but the order will not be created).
For transaction flow, it should work, as we have there single transaction block, but here I would also expect on_commit. I mean, converting checkout into order is our main action for checkout-complete. Saving address is an additional action that should not have a potential impact on rolling back the whole order creation process.

Reply:
I'll create a ticket for that.